luigi/task.py

```diff
@@ -484,6 +484,15 @@ def complete(self):
 
         return all(itertools.imap(lambda output: output.exists(), outputs))
 
+    @classmethod
+    def bulk_complete(cls, parameter_tuples):
+        """Returns those of parameter_tuples for which this Task is complete.
+
+        Override (with an efficient implementation) for efficient scheduling
+        with range tools. Keep the logic consistent with that of complete().
+        """
+        raise NotImplementedError
```

Comment:
actually nvm about #2 – probably a bad idea to creates lots of Task objects on the fly